1103 cherry pick improvements from tutorial

kykrueger opened the pull request:
## Purpose of the Change and Brief Change Log
Cherry picks upstream changes discussed in https://github.com/nebulastream/nebulastream/issues/1103 from the tutorial branch.

Agreed changes are 
- feat(SingleNodeWorker): Handle Termination Signal
- feat(Sinks): Adds JSON Format

## Verifying this change

Handle termination signal and the addition of the JSON format are without tests.
I assume a JSON input and output test should still be done before merging this.

For termination, a test may also still be added.

## What components does this pull request potentially affect?
- NodeEngine
- SingleNodeWorkerStarter
- PrintSink
- FileSink

## Documentation
- No new documentation included.

## Issue Closed by this pull request:
Closes https://github.com/nebulastream/nebulastream/issues/1103

ls-1801 reviewed (changes requested):
re-request once ci is green(ish)

kykrueger commented:
Ah, that makes sense. I'll put them back

kykrueger commented:
@ls-1801 I squashed all of the clang-tidy changes as discussed.
I've added some error checking when setting the signal_handler for one of the clang-tidy warnings.

The remaining warning about declaring the semaphore in the global scope without exception handling seems to be typical when it comes to using semaphores. Not sure if it is worth changing.

kykrueger commented:
@alepping @keyseven123 , could I get a second review?
